keps/993-two-phase-admission/README.md

```diff
@@ -253,6 +254,42 @@ The controller implementing a particular check should:
 * After approving the workload, keep an eye on the check if it starts failing,
   fail the check and cause the workload to move back to the suspended state.
 
+### Preemption Admission Check Controller
+
+In this proposal, the time to evict the preemption candidates varies based on the preemptor state.
+The scheduler will not issue the eviction during it's process instead it will set a `Pending` admission check
+that is manged by a single instance of a new built-in admission check controller.
+
+The **Preemption Admission Check Controller** will:
+
+- Watch for a change in state of the workloads pending preemption.
+- Watch for workloads that are finishing execution and therefore releasing quota.
+- Watch for AdmissionCheck changes, since the preemption policy can change.
+
+The preemption controller uses the kueue cache, since it needs to check the state of workloads admitted to the ClusterQueues.
+
+At every run the controller will get the list of workloads pending preemption.
+
+The workloads pending preemption are divided into:
+- `preemtingLeter` - Workloads having at least one check that uses AfterCheckPassedOrOnDemand policy with the state `pending`.
+- `preemtingNow` - Workloads that expect to be able to issue evictions or potentially change their preemption state in the current cycle.
+
+Then:
+1. Remove all workloads from the snapshot.
+2. For every workload in `preemtingNow` , in the order of their priority:
+  - If it can fit without the need of evicting other workloads
+    - Set its admission check to `Ready`
+    - Add it to the snapshot
+  - If it cannot fit
+    - Get an updated list of eviction candidates
+      - If the updated list is not empty.
+        - Issue the eviction to the candidates.
+        - Add it to the snapshot
+      - If the updated list is empty, meaning the preemption cannot be done.
+        - Set its admission check to `Retry`, the quota reservation will be lost and the workload placed in the queue waiting for a new QuotaReservation.
+
+**NOTE** The list of candidates is picked out from the list of workloads holding a QuotaReservation, regardless if they are fully Admitted or not.
```

Comment:
Is QuotaReservation enough? Consider jobs waiting for ProvReq to be fulfilled could be waiting for hours. There is no point in preempting them yet.

Reply:
The ProvRequest Admission check can be configured with `AfterCheckPassedOrOnDemand` policy, the evictions will be done only when the provisioning is finished (and only if still needed).

In the future , if we can know that the provisioning request has no chance to succeed for long period of times, the ProvRecAC can set its state to Retry this will fully release the quota.After that time the state can be set to Pending to put wl back in the queue.

Reply:
My question is more about the candidates for eviction. Obviously, fully Admitted workloads are candidates for eviction.
Other than that, I think it should be only the ones that have checks ready or are requesting preemption themselves.